### PR TITLE
Cross build Scala 2.11 and 2.12

### LINKED ISSUE #18: Support both Scala 2.11 and 2.12
We run Zeppelin 0.8.2 in our infrastructure and the Spark interpreter in that version doesn't support Scala 2.12, there is a fix in place that won't be released until Zeppelin 0.9.0 (https://jira.apache.org/jira/browse/ZEPPELIN-3552)

To support Zeppelin we build out everything, including spark from source, to use Scala 2.11

Currently there is no documented way to build the Delta connectors using Scala 2.11 outside of potentially modifying the variable set here: https://github.com/delta-io/connectors/blob/master/build.sbt#L31

### PR DESCRIPTION
**Pull request by zsxwing**
Add build support for Scala 2.11.

Closes #18

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -8,4 +8,5 @@ jobs:
       - run:
           name: Run tests 
           command: |
-            build/sbt coreTest/test hive/test
+            build/sbt "+ coreTest/test"
```
Comment (zsxwing) [inline .circleci/config.yml]: Tried but didn't find any way to merge these two commands into one SBT command :(

File: README.md [hunk 2]
```diff
@@ -48,7 +55,7 @@ Before starting your Hive CLI or running your Hive script, add the following spe
 </property>
 ```
 
-The second step is to upload the above two JARs to the machine that runs Hive. Finally, add the paths of the JARs toHive’s environment variable, `HIVE_AUX_JARS_PATH`. You can find this environment variable in the `hive-env.sh` file, whose location is ``/etc/hive/conf/hive-env.sh` on an EMR cluster. This setting will tell Hive where to find the connector JARs.
+The second step is to upload the above two JARs to the machine that runs Hive. Finally, add the paths of the JARs toHive’s environment variable, `HIVE_AUX_JARS_PATH`. You can find this environment variable in the `hive-env.sh` file, whose location is `/etc/hive/conf/hive-env.sh` on an EMR cluster. This setting will tell Hive where to find the connector JARs.
```
Comment (zsxwing) [inline README.md]: Remove an extra `` ` ``

File: build.sbt [hunk 3]
```diff
@@ -92,8 +94,13 @@ lazy val assemblySettings = Seq(
     case "reference.conf" => MergeStrategy.concat
     case _ => MergeStrategy.first
   },
+  assemblyJarName in assembly := s"${name.value}-assembly_${scalaBinaryVersion.value}-${version.value}.jar",
```
Comment (zsxwing) [inline build.sbt]: Add the scala version to the assembly jar

File: build.sbt [hunk 4]
```diff
@@ -92,8 +94,13 @@ lazy val assemblySettings = Seq(
...
 
-  assemblyShadeRules in assembly := Seq(
+  assemblyShadeRules in assembly :=
+    (if (scalaBinaryVersion.value == "2.11") Seq(
```
Comment (zsxwing) [inline build.sbt]: Scala 2.11 compiler doesn't happen when sharding json4s :(

File: coreTest/src/main/scala/Test.scala [hunk 5]
```diff
@@ -18,12 +18,14 @@ package test
 
 import shadedelta.io.delta.tables._
 
+import java.nio.file.Files
+
+import scala.collection.JavaConverters._
+
 import org.apache.spark.sql._
 import org.apache.spark.sql.functions._
-// import shadedelta.org.apache.commons.io.FileUtils
```
Comment (zsxwing) [inline coreTest/src/main/scala/Test.scala]: This one is not in the shaded jar. So I changed to use `Files` to clean up files. I just realized this test leaks tons of files in my laptop :)

File: build.sbt (hunk 4 above)
Comment (tdas) [reply build.sbt]: what is the error?

Comment (zsxwing) [reply build.sbt]:
```
hive/src/main/scala/io/delta/hive/DeltaStorageHandler.scala:189: Symbol 'term org.json4s' is missing from the classpath.
[error] This symbol is required by 'method org.apache.spark.sql.types.DataType.jsonValue'.
```
The above error when compiling codes. Looks like the compiler reads the original `org.json4s` name from the metadata of `method org.apache.spark.sql.types.DataType.jsonValue` and tries to load it.

**Comment by tdas** [comment]
LGTM.

**Pull request merged**